api/controllers/FileUploadController.js

```diff
@@ -61,34 +69,28 @@ const FileUploadController = {
         return multer(multerOptions).array(FORM_INPUT_NAME);
     },
 
-    _clearExistingErrorMessages(req) {
-        req.session.eApp.uploadMessages.errors = [];
-        req.session.eApp.uploadMessages.fileCountError = false;
-        req.session.eApp.uploadMessages.infectedFiles = [];
-        req.session.eApp.uploadMessages.noFileUploadedError = false;
-    },
-
     async _errorChecksAfterUpload(req, res, err) {
         const hasNoFiles = req.files.length === 0;
 
         if (hasNoFiles) {
-            req.session.eApp.uploadMessages.noFileUploadedError = true;
+            req.flash('errors', ['noFileUploadedError']);
             sails.log.error('No files were uploaded.');
             FileUploadController._redirectToUploadPage(res);
             return;
         }
 
-       removeFilesIfLarge(req);
+        removeFilesIfLarge(req);
 
         if (err) {
             const fileLimitExceeded = err.code === MULTER_FILE_COUNT_ERR_CODE;
             if (fileLimitExceeded) {
-                req.session.eApp.uploadMessages.fileCountError = true;
+                req.flash('errors', ['fileCountError']);
```

Comment:
you could use the codes for req.flash. eg `req.flash('errors', ['err.code'])`. It will keep the errors synced up as well, so you don't have to manage separate names for errors (codes and string).

 Although it works better with typescript + enums with intellisense.

Reply:
hmm, does this make sense if I'm only handling one error code?